### PR TITLE
Issue 65

### PR DESCRIPTION
**Pull request by devknight216**
## Pull Request type

- [x] Feature

## What is the current behavior?

Issue Number: 65

## What is the new behavior?

- Tags are saved once the note/audio... is saved.
- TypeAhead for tags is now enabled. Tag list is from the saved tags.

## Does this introduce a breaking change?

- [x] No

### CODE REVIEW

File: src/components/dialogs/AddEvidenceDialog.vue [hunk 1]
```diff
@@ -400,7 +401,31 @@ export default {
       hotkeys: "config/hotkeys",
       config: "config/fullConfig",
       credentials: "auth/credentials",
+      configTags: "config/tags",
+      sessionItems: "sessionItems",
     }),
+    filteredTags() {
```
Comment (dacoaster) [inline src/components/dialogs/AddEvidenceDialog.vue:407]: Please calculate this each time the dialog is displayed and then save it so we don't recompute it many times.

File: src/components/dialogs/EditEvidenceDialog.vue [hunk 2]
```diff
@@ -360,7 +361,31 @@ export default {
       hotkeys: "config/hotkeys",
       config: "config/fullConfig",
       credentials: "auth/credentials",
+      configTags: "config/tags",
+      sessionItems: "sessionItems",
     }),
+    filteredTags() {
```
Comment (dacoaster) [inline src/components/dialogs/EditEvidenceDialog.vue:367]: Same as above.

File: src/components/dialogs/NoteDialog.vue [hunk 3]
```diff
@@ -285,6 +287,32 @@ export default {
     };
   },
   computed: {
+    ...mapGetters({
+      configTags: "config/tags",
+      sessionItems: "sessionItems",
+    }),
+    filteredTags() {
```
Comment (dacoaster) [inline src/components/dialogs/NoteDialog.vue:294]: Same as above.

File: src/components/settings/TagsTab.vue [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+              placeholder="Add tag"
+              @input="
+                (evt) =>
+                  $store.commit('config/editTagText', {
+                    id: tag.id,
+                    text: evt.target.value,
+                  })
+              "
+              :value="tag.text"
+            />
```
Comment (dacoaster) [inline src/components/settings/TagsTab.vue:22]: Please make this a `vue-tags-input` that they can enter tags directly into.  (Without autocomplete.)

File: src/modules/PersistenceUtility.js [hunk 5]
```diff
@@ -34,6 +34,7 @@ const defaultConfig = {
   ai: {
     enabled: false,
   },
+  tags: [],
```
Comment (dacoaster) [inline src/modules/PersistenceUtility.js:37]: Please change the name to `defaultTags`.

File: src/store/modules/config.js [hunk 6]
```diff
@@ -16,6 +16,7 @@ export const config = {
     debugMode: false,
     summary: false,
     templates: [],
+    tags: [{ id: "1", text: "Canada" }],
```
Comment (dacoaster) [inline src/store/modules/config.js:19]: Please change to `defaultTags`.

File: src/store/modules/config.js [hunk 7]
```diff
@@ -114,6 +131,7 @@ export const config = {
     fullConfig: (state) => {
       return state;
     },
+    tags: (state) => state.tags,
```
Comment (dacoaster) [inline src/store/modules/config.js:134]: Please change to `defaultTags`.

File: src/views/NoteEditorView.vue [hunk 8]
```diff
@@ -147,6 +149,32 @@ export default {
     });
   },
   computed: {
+    ...mapGetters({
+      configTags: "config/tags",
+      sessionItems: "sessionItems",
+    }),
+    filteredTags() {
```
Comment (dacoaster) [inline src/views/NoteEditorView.vue:156]: Same comment as above on calculating this once.

**Review by dacoaster — CHANGES_REQUESTED** [review]

File: src/components/ControlPanel.vue [hunk 9]
```diff
@@ -733,6 +734,7 @@
       <AddEvidenceDialog
         v-if="evidenceData"
         v-model="addEvidenceDialog"
+        :evidenceData="evidenceData"
```
Comment (dacoaster) [inline src/components/ControlPanel.vue:737]: @devknight216 - This looks like a duplicate of the line below and doesn't exist in the `props` of `AddEvidenceDialog` so please remove it

**Review by dacoaster — CHANGES_REQUESTED** [review]
@devknight216 this needs one small update and then its ready to merge

File: src/components/ControlPanel.vue (hunk 9 above)
Comment (devknight216) [reply src/components/ControlPanel.vue]: @dacoaster, just removed that. :) Please merge it on your side.

**Review by dacoaster — APPROVED** [review]

**Pull request merged**